gitclone/checkout_simple_strategy.go

```diff
@@ -0,0 +1,129 @@
+package gitclone
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/bitrise-io/bitrise-init/step"
+	"github.com/bitrise-io/go-utils/command/git"
+)
+
+type checkoutStrategy interface {
+	Validate() error
+	Do(gitCmd git.Git) *step.Error
```

Comment:
I think Do should have a strategy independent options (depth, tags, and so on) as input.

As these options are independent of the strategy from user's point of view.